src/test/java/com/alphawallet/attestation/demo/VerifyAttestation.sol

```diff
@@ -0,0 +1,793 @@
+    {
+        uint length = uint8(byteCode[decodeIndex++]);
+        bytes32 store = 0;
+        for (uint j = 0; j < length; j++) store |= bytes32(byteCode[decodeIndex++] & 0xFF) >> (j * 8);
+        objCodes[objCodeIndex++] = uint256(store);
+        Status memory retVal;
+        retVal.decodeIndex = decodeIndex;
+        retVal.objCodeIndex = objCodeIndex;
+
+        return retVal;
+    }
+
+    struct Status {
+        uint decodeIndex;
+        uint objCodeIndex;
+    }
+
+    function decodeObjectIdentifier(bytes memory byteCode, uint256[] memory objCodes, uint objCodeIndex, uint decodeIndex) private pure returns(Status memory)
+    {
+        uint length = uint8(byteCode[decodeIndex++]);
+
+        Status memory retVal;
+
+        //1. decode leading pair
+        uint subIDEndIndex = decodeIndex;
+        uint256 subId;
+
+        while ((byteCode[subIDEndIndex] & 0x80) == 0x80)
+        {
+            require(subIDEndIndex < byteCode.length);
+            subIDEndIndex++;
+        }
+
+        uint subidentifier = 0;
+        for (uint i = decodeIndex; i <= subIDEndIndex; i++)
+        {
+            subId = uint256(uint8(byteCode[i] & 0x7f)) << ((subIDEndIndex - i) * 7);
+            subidentifier |= subId;
+        }
+
+        if (subidentifier < 40)
+        {
+            objCodes[objCodeIndex++] = 0;
+            objCodes[objCodeIndex++] = subidentifier;
+        }
+        else if (subidentifier < 80)
+        {
+            objCodes[objCodeIndex++] = 1;
+            objCodes[objCodeIndex++] = subidentifier - 40;
+        }
+        else
+        {
+            objCodes[objCodeIndex++] = 2;
+            objCodes[objCodeIndex++] = subidentifier - 80;
+        }
+
+        subIDEndIndex++;
+
+        while (subIDEndIndex < (decodeIndex + length) && byteCode[subIDEndIndex] != 0)
+        {
+            subidentifier = 0;
+            uint256 subIDStartIndex = subIDEndIndex;
+
+            while ((byteCode[subIDEndIndex] & 0x80) == 0x80)
+            {
+                require(subIDEndIndex < byteCode.length);
+                subIDEndIndex++;
+            }
+            subidentifier = 0;
+            for (uint256 j = subIDStartIndex; j <= subIDEndIndex; j++)
+            {
+                subId = uint256(uint8(byteCode[j] & 0x7f)) << ((subIDEndIndex - j) * 7);
+                subidentifier |= subId;
+            }
+            objCodes[objCodeIndex++] = subidentifier;
+            subIDEndIndex++;
+        }
+
+        decodeIndex += length;
+
+        retVal.decodeIndex = decodeIndex;
+        retVal.objCodeIndex = objCodeIndex;
+
+        return retVal;
+    }
+
+    function endContract() public payable
+    {
+        if(msg.sender == owner)
+        {
+            selfdestruct(owner);
+        }
+        else revert();
+    }
+    
+    //Most optimal 4 byte array variable length concatenate. If sizes of the inputs are static we can optimise this even further
+    function concat4(
+        bytes memory _bytes1,
```

Comment:
These are in fact static since they are all elliptic curve points. So maybe we can optimise a bit more here?